tests/lib.rs

```diff
@@ -173,13 +175,14 @@ fn write_random_sized_out_of_sequence_writes_with_gaps_and_overlaps() {
     let mut se = SelfEncryptor::new(&mut my_storage, datamap::DataMap::None);
     for ind in index {
       let piece_size : usize = pieces[ind].len();
-      let offset : usize = ind * piece_size;
-      total_size = std::cmp::max(total_size, offset + piece_size);
+      let offset : usize = offsets[ind];
+      total_size = std::cmp::max(total_size, offset+ piece_size);
       se.write(pieces[ind], offset as u64);
-
-
+      assert!(DATA_SIZE >= total_size as u64);
+      let decrypted = se.read(offset as u64, piece_size as u64);
+      assert_eq!(decrypted, original[offset..(offset + piece_size)].to_vec());
     }
+    //assert_eq!(total_size as u64, se.len());
```

Comment:
This slipped into this PR, but I'd need to clean it up.  The example PR is not indented to touch tests/lib.rs.  Will merge latest master